src/qililab/circuit_transpiler/circuit_transpiler.py

```diff
@@ -12,72 +12,168 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Class that translates a Qibo Circuit into a PulseSequence"""
+"""Circuit Transpiler class"""
+
+
 import contextlib
 from dataclasses import asdict
+from typing import TYPE_CHECKING
 
 import numpy as np
+from qibo import gates
 from qibo.gates import Gate, M
-from qibo.models.circuit import Circuit
+from qibo.models import Circuit
 
 from qililab.chip.nodes import Coupler, Qubit
+from qililab.circuit_transpiler.gate_decompositions import translate_gates
 from qililab.constants import RUNCARD
 from qililab.instruments import AWG
-from qililab.platform import Bus, Platform
 from qililab.settings.gate_event_settings import GateEventSettings
-from qililab.transpiler import Drag
 from qililab.typings.enums import Line
-from qililab.utils import Factory, qibo_gates
+from qililab.utils import Factory
 
-from .pulse import Pulse
-from .pulse_event import PulseEvent
-from .pulse_schedule import PulseSchedule
+from ..pulse.pulse import Pulse
+from ..pulse.pulse_event import PulseEvent
+from ..pulse.pulse_schedule import PulseSchedule
+from .native_gates import Drag, Wait
 
+if TYPE_CHECKING:
+    from qililab.platform import Bus, Platform  # pragma: no cover
 
-class CircuitToPulses:  # pylint: disable=too-few-public-methods
-    """Translates a list of circuits into a list of pulse sequences (each circuit to an independent pulse sequence)
-    For each circuit gate we look up for its corresponding gates settings in the runcard (the name of the class of the circuit
-    gate and the name of the gate in the runcard should match) and load its schedule of GateEvents.
-    Each gate event corresponds to a concrete pulse applied at a certain time w.r.t the gate's start time and through a specific bus
-    (see gates settings docstrings for more details).
 
-    Measurement gates are handled in a slightly different manner. For a circuit gate M(0,1,2) the settings for each M(0), M(1), M(2)
-    will be looked up and will be applied in sync. Note that thus a circuit gate for M(0,1,2) is different from the circuit sequence
-    M(0)M(1)M(2) since the later will not be necessarily applied at the same time for all the qubits involved.
+class CircuitTranspiler:
```

Comment:
I would still rename this to `QiboCircuitTranspiler` since it is a library-specific transpiler and to avoid future breaking issues if we add our own circuit class.

Reply:
I agree if indeed at some time we introduce a very different circuit structure. However for circuit representations like qasm it would be pretty easy to adapt the current `CircuitTranspiler` for that as well / instead of qibo. Since we don't know yet what we'll do I'm not too sure about the name change but we can comment this with the rest of the team and see what they think.